### PR TITLE
Add advanced conflation options to Create Differential Changeset

### PR DESCRIPTION
**Pull request by brianhatchl**
refs #1685

### CODE REVIEW

**Comment by brianhatchl** [comment]
The UI for Derive Differential Changeset now offers access to Advanced Options

image

File: css/hoot/components/_clip.scss [hunk 1]
```diff
@@ -173,8 +173,16 @@
         padding-right: 10px;
     }
 
-    input[type="text"] {
+/*    input[type="text"] {
         border: none;
         padding: 0;
+    }*/
```
Comment (brianhatchl) [inline css/hoot/components/_clip.scss]: Didn't think we needed this and it was messing up my advanced options

File: modules/Hoot/tools/differentialChangeset.js [hunk 2]
```diff
@@ -5,31 +5,98 @@ import { formatBbox } from './utilities';
...
-    render() {
+    async render() {
         let titleText = this.instance.bboxSelectType === 'visualExtent'
             ? 'Create Differential from Visual Extent'
             : this.instance.bboxSelectType === 'boundingBox'
                 ? 'Create Differential from Bounding Box'
                 : 'Create Differential';
 
+        this.advOpts = await Hoot.api.getAdvancedOptions('differential');
+        let advForm = this.advOpts.map(this.formFactory.advOpt2DomMeta);
```
Comment (brianhatchl) [inline modules/Hoot/tools/differentialChangeset.js:19]: Get advanced options for Differential conflate

File: modules/Hoot/ui/modals/changesetStats.js [hunk 3]
```diff
@@ -70,11 +72,18 @@ export default class ChangesetStats {
...
         this.createTable();
+
+        this.updateSubmitButton();
+    }
+
+    updateSubmitButton() {
+        this.submitButton.attr( 'disabled', function() {
+                var n = d3.select('#preset-input-comment').node();
+                return (n && n.value.length) ? null : true;
+            });
```
Comment (brianhatchl) [inline modules/Hoot/ui/modals/changesetStats.js:86]: make sure submit button is disabled if no comment has been entered

File: modules/Hoot/ui/modals/importDataset.js [hunk 4]
```diff
@@ -81,6 +81,9 @@ export default class ImportDataset {
 
         //Add advanced options to form
         this.advOpts = await Hoot.api.getAdvancedImportOptions();
+        this.advOpts.forEach(opt => {
+            opt.hidden = true;
+        });
```
Comment (brianhatchl) [inline modules/Hoot/ui/modals/importDataset.js:86]: formFactory doesn't hide by default anymore

File: modules/Hoot/ui/sidebar/advancedOpts.js [hunk 5]
```diff
@@ -225,7 +225,6 @@ export default class AdvancedOpts {
             .classed( 'combobox-caret', d => d.members.length );
 
         caretWrap.merge(caretWrapEnter);
-        caretWrap.on( 'click', instance.showBody);
```
Comment (brianhatchl) [inline modules/Hoot/ui/sidebar/advancedOpts.js:228]: this was redundant and seemed to cause #1686

**Review by bobi7 — APPROVED** [review]
💯

**Pull request merged**